zerver/actions/create_user.py

```diff
@@ -216,9 +216,11 @@ def add_new_user_history(user_profile: UserProfile, streams: Iterable[Stream]) -
         # Create UserMessage rows for the backfill.
         ums_to_create = []
         for message_id in backfill_message_ids:
-            um = UserMessage(user_profile=user_profile, message_id=message_id)
+            um = UserMessage(
+                user_profile=user_profile, message_id=message_id, flags=UserMessage.flags.historical
```

Comment:
Why is adding the `historical` flag correct? Probably that should be in the commit message. Reasons in favor:
- It's not a message you actually received, so semantically that's right.
- I assume it will avoid annoying "you subscribed" notices?

But I guess I'm worried whether this has impact on:
- Whether those messages will show up in your search results.
- Whether those messages show up in "Combined feed"

Reply:
Updated the commit message.

I checked by testing manually:
* Messages with "historical" flag appear in Combined Feed, recent messages etc
* Search works 

I checked the use of "historical" flag in code too.. The only place where we exclude `historical` flagged messages is 
```
get_mentions_for_message_updates()

 # We exclude UserMessage.flags.historical rows since those
 # users did not receive the message originally, and thus
 # probably are not relevant for reprocessed alert_words,
 # mentions and similar rendering features.  This may be a
 # decision we change in the future.
```
and I think  it's fine because of the reason mentioned in the comment.

Reply:
OK that sounds correct to me!